src/plonk/circuit.rs

```diff
@@ -1062,14 +1062,14 @@ impl<F: Field> ConstraintSystem<F> {
     pub fn enable_constant(&mut self, column: Column<Fixed>) {
         if !self.constants.contains(&column) {
             self.constants.push(column);
-            self.enable_equality(column.into());
+            self.enable_equality(column);
         }
     }
 
     /// Enable the ability to enforce equality over cells in this column
-    pub fn enable_equality(&mut self, column: Column<Any>) {
+    pub fn enable_equality<C: Into<Column<Any>> + Copy>(&mut self, column: C) {
```

Comment:
The `Into` bound is useful as it expands the set of valid arguments. By contrast, I don't think the `Copy` bound should be present, as it restricts the set of valid arguments by preventing non-`Copy` values from being moved in. We do not need the ability to copy inside this method.

The given rationale also doesn't make sense to me:

> The `Copy` is added since Into consumes the `Column`, and sometimes we
fall into the situation of needing to clone when we can simply require
the copy and avoid more code verbosity.

If a caller is in a situation where with they need to clone, then adding `Copy` does not help them; it just replaces one error (value is moved) with another (type doesn't impl `Copy`).

Reply:
I see.

I basically took that direction since all of the types that are converted into `Column<Any>` impl `Copy`. Therefore, made sense to me. 

But makes sense since it adds restrictions to the trait object that are not necessary.

Thanks!